main/plugins/org.talend.designer.bigdata.di/src/main/java/org/talend/designer/bigdata/di/components/DesignerDIComponent.java

```diff
@@ -0,0 +1,53 @@
+package org.talend.designer.bigdata.di.components;
+
+import org.jetbrains.annotations.NotNull;
+import org.talend.core.model.process.ElementParameterParser;
+import org.talend.core.model.process.INode;
+import org.talend.designer.codegen.config.CodeGeneratorArgument;
+
+import java.util.Optional;
+
+public interface DesignerDIComponent {
+
+    String name();
+
+    interface BigDataDIComponent extends DesignerDIComponent {
+
+        CodeGeneratorArgument codeGeneratorArgument();
+        @NotNull
+        static <T> T parameter(final INode node, final String code, @NotNull T defaultValue) throws ClassCastException {
+            return Optional.ofNullable(ElementParameterParser.getObjectValue(node, code))
+                    .map(parameter -> (T) parameter)
+                    .orElse(defaultValue);
+        }
+        @NotNull
+        default <T> T parameter(final String code, @NotNull T defaultValue) throws ClassCastException {
+            return parameter(((INode) codeGeneratorArgument().getArgument()), code, defaultValue);
+        }
+
+        static boolean getBooleanParameter(final INode node, final String string){
+            return Optional.ofNullable(ElementParameterParser.getBooleanValue(node, string))
+                    .orElse(false);
+        }
+
+        @Override
+        default String name() {
+            return ((INode) codeGeneratorArgument().getArgument()).getUniqueName() + "Component";
+        }
+
+        default String getCid() {
```

Comment:
```suggestion
        default String cid() {
```
Other getters doesn't have get prefix so that's to be consistent

Reply:
I have changes other getters, so now they are consistent